model-optimizer/mo/ops/slice.py

```diff
@@ -142,8 +142,6 @@ def infer(node: Node):
             slice_idx[axes[i]] = slice(starts[i], ends[i], steps[i])
         if input_value is None or any(is_dynamic_slice(s) for s in slice_idx):
             output_shape = get_shape_from_slice(input_shape, slice_idx)
-            if np.ma.any(output_shape <= 0):
```

Comment:
Shoud we update nGraph slice operation ipmlementation too? And Slice specification?

Reply:
No, we shouldn't since ngraph and reference behave correctly, with support of 0-dimensins. 
More specifically, in Slice/StridedSlice specification it's written that out of bound values are silently clamped. According this, when out of bound values met the following should happen.
```
inp.shape = [1 3 225 225]
res = StridedSlice(inp, begin=[300], end=[301], axes=[2]) =...
StridedSlice(inp, begin=[225], end=[225], axes=[2])
res.shape = [1 3 0 225] since shape[2] = end-begin
```
I checked manually that ngraph SS/Slice core and reference implementations match this behavior.